docs/reference/mapping/types/runtime.asciidoc

```diff
@@ -0,0 +1,585 @@
+[[runtime]]
+=== Runtime
+Typically, you must index fields to {es} before they can be retrieved,
+aggregated, or searched. With _runtime fields_ beta[], you can explicitly
+define a field in the mapping and access it at search time without indexing
+your data during ingest time.
+
+Because runtime fields aren't indexed, you can more quickly ingest raw data
+into the Elastic Stack and access it immediately. Runtime fields are accessible
+from the search API like any other field. The <<search-field-caps,field capabilities API>> returns runtime fields as both searchable and aggregatable.
+
+By dynamically evaluating runtime fields at search time, you can optimize disk
+space by choosing which fields to index. If you later decide that a runtime
+field is increasingly used for filtering or aggregation, you can add the field
+to `_source` to gain faster search speed.
+
+Alternatively, you can specify a `runtime_mappings` section in a search request
+to use runtime fields that exist only as part of the query. This
+flexibility allows you to create or override field values in `_source` for the
+duration of the query without modifying the field itself.
+
+Runtime fields help to alleviate several common issues when using {es}:
+
+* Reindexing your data between development iterations is slow and can make
+experimenting on large datasets difficult
+* Indexing data before searching makes running one-off searches costly and
+resource intensive
+* Indexing all of your data instead of just the fields you want to search
+requires more disk space to gain performance at search time
+* Reindexing data for time-based indices to ensure that existing indices
+include any new fields in the index template is slow
+* Determining how a field is evaluated in {kib} at index or search time is
+difficult for scripted fields because they have different needs based on where
+they are implemented
+
+Runtime fields incur performance costs at search time, depending
+on the <<runtime-params-runtime-type,`runtime_type`>>. For example, let's say
+you created an anomaly detection job that operates on the `@timestamp` field.
+If the `@timestamp` field is a runtime field, the search cost would be extremely
+high because the data isn't indexed and {es} must compute the value for each
+document that matches the query.
+
+NOTE: Computing values for runtime fields in each document that might match a
+query impacts search speed. Use the <<async-search,asynchronous search API>>
+to run searches that include runtime fields.
+
+Runtime fields are useful when working with log data, especially when you're
+unsure about the data structure. Your search speed decreases, but your index
+size is much smaller and you can more quickly process logs without having to
+index them.
```

Comment:
I think a recipe for how to extract things from log data might be better at explaining it then a paragraph.

Reply:
The [Examples](https://elasticsearch_62653.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/runtime.html#runtime-examples) section shows how to extract stuff from log data. Is it enough to add a link from this paragraph to that section?

Reply:
I guess my question is, do we need the paragraph at all then? If we have examples that show, maybe we don't have to tell?

Reply:
I still think the information is relevant, especially since I've moved this bit into a separate section for [Use cases](https://elasticsearch_62653.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/runtime.html#runtime-use-cases). I can add a link to the Examples section from there. Also still need to incorporate changes for those bulleted items.